### DISCUSSION TITLE
ScriptReference/SerializedProperty.NextVisible

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#168] [Notes]
# ScriptReference/SerializedProperty.NextVisible

https://docs.unity3d.com/ScriptReference/SerializedProperty.NextVisible.html

**Comment by vertxxyz** [comment]
To iterate over the children of a Property for the purposes of displaying PropertyFields:
```csharp
SerializedProperty end = prop.GetEndProperty();
bool enterChildren = true;
while (prop.NextVisible(enterChildren) && !SerializedProperty.EqualContents(prop, end))
{
    root.Add(new PropertyField(prop));
    enterChildren = false;
}
```
Where prop is your SerializedProperty.
> **Warning**
> Using the `NextVisible` iterator will advance your SerializedProperty, so use [`Copy`](https://docs.unity3d.com/ScriptReference/SerializedProperty.Copy.html) if you want to make a copy to iterate over.
